### PR TITLE
Add generic ConservativeDuDt function struct

### PR DESCRIPTION
**Pull request by wthrowe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Evolution/Conservative/ConservativeDuDt.hpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#include "NumericalAlgorithms/LinearOperators/Divergence.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \ingroup ConservativeGroup
+/// \brief Calculate \f$\partial u/\partial t\f$ for a conservative system.
+///
+/// The time evolution of the variables \f$u\f$ of any conservative
+/// system is given by \f$\partial_t u = - \partial_i F^i + S\f$,
+/// where \f$F^i\f$ are the fluxes as \f$S\f$ are the sources.
```
Comment (mar-celine) [inline src/Evolution/Conservative/ConservativeDuDt.hpp]: Where F are the fluxes and S are the sources. (no "as")

File: src/Evolution/Conservative/ConservativeDuDt.hpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        const gsl::not_null<db::item_type<TensorTags>*>... dt_u,
+        const db::item_type<TensorTags>&... div_u,
+        const db::item_type<SourcedTags>&... sources) noexcept {
+      const auto negate = [](const auto a, const auto& b) noexcept {
+        for (size_t i = 0; i < a->size(); ++i) {
+          (*a)[i] = -b[i];
+        }
+        return nullptr;
+      };
+      expand_pack(negate(dt_u, div_u)...);
```
Comment (mar-celine) [inline src/Evolution/Conservative/ConservativeDuDt.hpp:43]: Should there be a useful error message for when a differing amount of dt_u's and div_u's are passed to this function?

**Review by mar-celine — APPROVED** [review]

**Comment by mar-celine** [comment]
@nilsdeppe look at this please

File: src/Evolution/Conservative/ConservativeDuDt.hpp (hunk 2 above)
Comment (wthrowe) [reply src/Evolution/Conservative/ConservativeDuDt.hpp:43]: You will get the usual compiler error for when you try to call a function with the wrong arguments.

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**